### PR TITLE
Fix 2p 1846 draft and some other 1846 bugs found 

### LINKED ISSUE #1754: 1846 2p Variant - in draft, companies should be paid for immediately
Current behavior inherits from the base 1846 behavior where payment is made at the end of the draft. In the 2p variant, this is unnecessary as the companies are not hidden.

A new draft class should be made that could be reused by other titles such as 18CZ and 18 Los Angeles (which is an 1846 “clone” but uses this style of draft for all player counts).

### LINKED ISSUE #1757: 1846 2p Variant - pair of ORs during draft need to be full ORs, not just privates paying
Current behavior is just for privates to pay, but it should be two full ORs, which includes Steamboat assignment and MS/BIG4 operating.

### LINKED ISSUE #1758: 1846 - second-to-last private drafted is shown
https://www.18xx.games/game/10959?action=28

So, the last private up for auction becomes public knowledge, but in this game, the player who drafted the second-to-last-one (the last hidden one) had that choice publicized.  While it is unlikely to be a surprise of what it is, this still makes public some information that should not be.

### LINKED ISSUE #1770: 1846 - draft - when a company is reduced to $10 and then passed on, the wrong player gets it
When a company in the draft is discounted to $10, if a player passes on it, the next player in priority order should become the owner, but the last player to pass gets the company.

https://www.18xx.games/game/11475?action=17 - michaeljb2 has the opportunity to buy Big 4 for $10; after passing, it should go to michaeljb but michaeljb2 gets it

https://www.18xx.games/game/11475?action=18

### LINKED ISSUE #1771: 1846 draft - if Big 4 is last company in auction, its price can go below $40
https://www.18xx.games/game/11475?action=17

### PR DESCRIPTION
**Pull request by michaeljb**
General improvements:

- sort private company payouts by owner (players first in priority order, then
  corporations in operating order), revenue, then name

1846 bug fixes:

- prevent Big 4 from being reduced beyond minimum price [Fixes #1771]
- don't show the name of the second-to-last private that was drafted [Fixes #1758]
- when a company costs $10, the player who passes on it does not get to buy it
  [Fixes #1770]
- return `[]` for `emergency_issuable_bundles` when the corporation does have a
  train; this hides the message about raising money by issuing shares in the buy
  trains view when the corporation has a train

1846 2p Variant improvements:

- add note in the log about removed corporation's second token

1846 2p Variant bug fixes:

- prevent first player in draft from passing
- companies are paid for immediately when drafted, better logging [Fixes #1754]
- perform two full ORs if both players pass [Fixes #1757]
    - don't call them X.1 and X.2, just 1 and 2
    - don't prompt user to assign steamboat if it's pointless (e.g., if it's a
      special OR in the draft round and no minors have been chosen by players)

Log sorting (Fixture game 3099, action 346), before:

```
MrDiskord collects $10 from Chicago and Western Indiana
B&O collects $20 from Tunnel Blasting Company
ERIE collects $15 from Meat Packing Company
GT collects $10 from Steamboat Company
ERIE collects $15 from Lake Shore Line
sirstevie3 collects $15 from Michigan Central
B&O collects $15 from Ohio & Indiana
```

After:

```
MrDiskord collects $10 from Chicago and Western Indiana
sirstevie3 collects $15 from Michigan Central
GT collects $10 from Steamboat Company
ERIE collects $15 from Lake Shore Line
ERIE collects $15 from Meat Packing Company
B&O collects $15 from Ohio & Indiana
B&O collects $20 from Tunnel Blasting Company
```

### CODE REVIEW

File: assets/app/view/game/round/operating.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def render
 
           if entity.player?
             left << h(Player, player: entity, game: @game)
-          elsif entity.corporation? || entity.minor?
+          elsif (entity.corporation? || entity.minor?) && entity.floated?
```
Comment (tobymao) [inline assets/app/view/game/round/operating.rb:44]: entity.operator?

File: lib/engine/game/base.rb [hunk 2]
```diff
@@ -643,9 +643,18 @@ def all_companies_with_ability(ability)
       end
 
       def payout_companies
-        @companies.select(&:owner).each do |company|
-          next unless (revenue = company.revenue).positive?
+        companies = @companies.select(&:owner).select { |c| c.revenue.positive? }
```
Comment (tobymao) [inline lib/engine/game/base.rb:646]: might as well just do what select with c.owner && c.revenu.positive

File: lib/engine/round/draft.rb [hunk 3]
```diff
@@ -12,6 +12,15 @@ def name
       def select_entities
         @game.players.reverse
       end
+
+      # def finished?
```
Comment (tobymao) [inline lib/engine/round/draft.rb:16]: comments

**Comment by michaeljb** [comment]
Addressed review comments in 68087699, now I'm running `validate.rb` locally

**Comment by tobymao** [comment]
ok looks good, i'll leave you to merge and pin

**Pull request merged**